fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraBaseResource.java

```diff
@@ -74,13 +75,15 @@ public FedoraResource getResourceFromPath(final String externalPath) {
     /**
      * Set the baseURL for JMS events.
      * @param uriInfo the uri info
+     * @param headers HTTP headers
      **/
-    protected void setUpJMSBaseURIs(final UriInfo uriInfo) {
+    protected void setUpJMSBaseURIs(final UriInfo uriInfo, final HttpHeaders headers) {
         try {
             final URI baseURL = uriInfo.getBaseUri();
             LOGGER.debug("setting baseURL = " + baseURL.toString());
             final ObservationManager obs = session().getWorkspace().getObservationManager();
-            final String json = "{\"baseURL\":\"" + baseURL.toString() + "\"}";
+            final String json = "{\"baseURL\":\"" + baseURL.toString() + "\","
+                    + "\"userAgent\":\"" + headers.getHeaderString("user-agent") + "\"}";
```

Comment:
Yes, and it would be better to build a JSON object instead of doing the string quoting ourselves.